File: app/src/main/java/net/tlalka/android/fiszki/listeners/MenuListener.java
package net.tlalka.android.fiszki.listeners;

import android.content.Context;
import android.content.Intent;
import android.view.View;
import android.view.View.OnClickListener;
import net.tlalka.android.fiszki.elements.PageElement;


public class MenuListener implements OnClickListener {

    private Context context;
    private PageElement pageElement;

    public MenuListener(Context context, PageElement pageElement) {
        this.context = context;
        this.pageElement = pageElement;
    }

    @Override
    public void onClick(View view) {
        context.startActivity(new Intent(context.getApplicationContext(), pageElement.getActivityClass()));
    }
}

File: app/src/main/kotlin/net/tlalka/fiszki/view/listeners/MenuListener.kt
package net.tlalka.fiszki.view.listeners

import android.content.Context
import android.content.Intent
import android.view.View
import android.view.View.OnClickListener

import net.tlalka.fiszki.view.elements.PageElement

class MenuListener(private val context: Context, private val pageElement: PageElement) : OnClickListener {

    override fun onClick(view: View) {
        context.startActivity(Intent(context.applicationContext, pageElement.getActivityClass()))
    }
}
